ast/compile.go

```diff
@@ -2707,9 +2710,13 @@ func newLocalVarGenerator(suffix string, node interface{}) *localVarGenerator {
 	return &localVarGenerator{exclude: exclude, suffix: suffix, next: 0}
 }
 
-func (l *localVarGenerator) Generate() Var {
+func (l *localVarGenerator) Generate(occ varOccurrence) Var {
+	prefix := "__local"
+	if occ == declaredVar {
```

Comment:
💭 I think it could be nice to have every type of occurrence be reflected in `Var`. I'd prefer different types (or a struct with an occurrence field) over encoding it in the variable name; but I suspect it's difficult to do that without breaking too many things.